lib/pageUtils.js

```diff
@@ -0,0 +1,24 @@
+export const scrollTo = (element, to, duration) => {
+  if (duration <= 0) return
+  const difference = to - element.scrollTop;
+  const perTick = difference / duration * 16
+
+  window.requestAnimationFrame(() => {
+    element.scrollTop = element.scrollTop + perTick
+    if (element.scrollTop === to) return
+    scrollTo(element, to, duration - 16)
+  })
+}
+
+export const urlQuery = ((a) => {
+  if (a == '') return {}
+  let b = {}
+  for (let i = 0; i < a.length; ++i) {
+    let p=a[i].split('=', 2)
+    if (p.length == 1)
+      b[p[0]] = ''
+    else
+      b[p[0]] = decodeURIComponent(p[1].replace(/\+/g, ' '))
+  }
+  return b
```

Comment:
This looks like excessive variable shortening, especially a/b. a will be the query right? So at least `q` to make it a little easier to understand
Then you'd probably want to reduce into b instead of the for loop